src/Checker/ReferenceAssignmentChecker.php

```diff
@@ -4,34 +4,113 @@
 namespace umulmrum\PhpReferenceChecker\Checker;
 
 
+use PhpParser\Node\Expr\AssignRef;
+use PhpParser\Node\Expr\MethodCall;
 use PhpParser\Node\Stmt\Class_;
 use PhpParser\ParserFactory;
 use Symfony\Component\Finder\Finder;
+use Symfony\Component\Finder\SplFileInfo;
+use umulmrum\PhpReferenceChecker\DataModel\MethodRepository;
+use umulmrum\PhpReferenceChecker\DataModel\NonReferenceAssignmentWarning;
 
 class ReferenceAssignmentChecker
 {
     /**
      * @param string $targetPath
      * @param string $classRepositoryPath
      *
-     * @return array
+     * @return NonReferenceAssignmentWarning[]
      */
     public function check($targetPath, $classRepositoryPath)
     {
         $this->init();
-        $this->buildMethodRepository($classRepositoryPath);
+        $repo = $this->buildMethodRepository($classRepositoryPath);
+        $warnings = $this->checkTargetPath($targetPath, $repo);
 
-        return [];
+        return $warnings;
     }
 
     private function init()
     {
         ini_set('xdebug.max_nesting_level', 3000);
     }
 
+    /**
+     * @param string           $targetPath
+     * @param MethodRepository $repository
+     *
+     * @return NonReferenceAssignmentWarning[]
+     */
+    private function checkTargetPath($targetPath, MethodRepository $repository)
+    {
+        $warnings = [];
+        $finder = new Finder();
+        if (is_file($targetPath)) {
```

Comment:
Please extract the handling for a single file into another method and call that method in the single-file case and from the finder loop.